libpages/server.go

```diff
@@ -519,9 +496,13 @@ func ListenAndServe(ctx context.Context,
 		IdleTimeout:       httpIdleTimeout,
 	}
 
-	httpRedirectServer := http.Server{
-		Addr:              ":80",
-		Handler:           http.HandlerFunc(server.redirectHandlerFunc),
+	httpServer := http.Server{
+		Addr: ":80",
+		// Enable http-01 by calling the HTTPHandler method, and set the
+		// fallback HTTP handler to nil, which means we'll get a handler that
+		// redirects all HTTP traffic that's not for ACME domain verification
```

Comment:
302 for GET/HEAD and 400 for other methods (since many user-agents always respond to a 302 with a GET request, so 302 only [predictably](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/307) works with GET/HEAD). ([source](https://github.com/golang/crypto/blob/13931e22f9e72ea58bb73048bc752b48c6d4d4ac/acme/autocert/autocert.go#L288-L295) and [comment](https://github.com/golang/crypto/blob/13931e22f9e72ea58bb73048bc752b48c6d4d4ac/acme/autocert/autocert.go#L248-L251).

I'll amend the comment.